pisi/operations/upgrade.py

```diff
@@ -207,6 +214,47 @@ def upgrade(A=[], repo=None):
     finally:
         ctx.exec_usysconf()
 
+def prepare_offline_upgrade(order):
+    # https://www.freedesktop.org/software/systemd/man/latest/systemd.offline-updates.html
+    #
+    # For a high-level overview
+    # 1. Fetch upgrades with eopkg up --fetch-only
+    # 2. Create symlink to /system-update to instruct systemd to perform offline upgrades
+    # 3. Soft reboot to initiate offline upgrades
+    # 3. eopkg-offline-upgrades.service will run eopkg up --bypass-update-repo --yes-all
+    # 4. System reboots and /system-update file is removed
+
+    # a bit of a hard coded safety hack here
+    if not os.path.exists('/usr/lib/systemd/system/system-update.target.wants/eopkg-offline-update.service'):
```

Comment:
Shouldn't this be checking for the normal service file, not the symlink?

Reply:
Or rather, both. As both should always exist on the system and the `system-update.target.wants` is only executed when offline updates are detected by the `/system-update` symlink?